test/integration/parachain/staging/setup/initialize.test.ts

```diff
@@ -187,4 +279,138 @@ describe("Initialize parachain state", () => {
+            existingCeiling
+        );
+
+        // set normal threshold
+        const setThresholdExtrinsic = sudoInterBtcAPI.api.tx.vaultRegistry.setSecureCollateralThreshold(
+            aUsdCcyPair,
+            encodedSecThresh
+        );
+
+        // set premium threshold
+        const setPremiumThresholdExtrinsic = sudoInterBtcAPI.api.tx.vaultRegistry.setPremiumRedeemThreshold(
+            aUsdCcyPair,
+            encodedPremThresh
+        );
+
+        // set liquidation threshold
+        const setLiquidationThresholdExtrinsic = sudoInterBtcAPI.api.tx.vaultRegistry.setLiquidationCollateralThreshold(
+            aUsdCcyPair,
+            encodedLiqThresh
+        );
+
+        // batch all
+        const batch = sudoInterBtcAPI.api.tx.utility.batchAll([
+            setCeilingExtrinsic,
+            setThresholdExtrinsic,
+            setPremiumThresholdExtrinsic,
+            setLiquidationThresholdExtrinsic,
+        ]);
+
+        const timeoutMs = 5 * APPROX_BLOCK_TIME_MS;
+        const [foundEvent] = await Promise.all([
+            waitForEvent(sudoInterBtcAPI, api.events.sudo.Sudid, false, timeoutMs),
+            sudoThis(batch),
+        ]);
+
+        assert.isTrue(
+            foundEvent,
+            `Cannot find Sudid event for setting thresholds in batch - timeout after ${timeoutMs} ms`
+        );
+    });
+
+    it("should fund and register aUSD foreign asset vaults", async function () {
+        // only get aUSD if it hasn't been set yet (e.g. when running this test in isolation)
+        !aUSD ? (aUSD = await getAUSDForeignAsset(sudoInterBtcAPI.assetRegistry)) : undefined;
+
+        if (aUSD === undefined) {
+            // no point in completing this if aUSD is not registered
+            this.skip();
+        }
+
+        // assign locally to make TS understand it isn't undefined
+        const aUsd: ForeignAsset = aUSD;
+
+        // free money to set with the help of sudo
+        const freeBalanceToSet = new MonetaryAmount(aUsd, 1000000000);
+        const collateralAmount = new MonetaryAmount(aUsd, 10000);
+        const currencyPair = newVaultCurrencyPair(api, aUsd, sudoInterBtcAPI.getWrappedCurrency());
+
+        const vaultAccountIdAndKeyrings: [KeyringPair, AccountId][] = [vault_1, vault_2, vault_3].map((keyringPair) => {
+            return [keyringPair, accountIdFromKeyring(keyringPair)];
+        });
+
+        // register the vaults with aUSD collateral
+        for (const [vaultKeyringPair, vaultAccountId] of vaultAccountIdAndKeyrings) {
+            const balanceTimeout = 5 * APPROX_BLOCK_TIME_MS;
+            // set balance and wait for event
+            const [balanceEventFound] = await Promise.all([
+                waitForEvent(sudoInterBtcAPI, api.events.tokens.BalanceSet, false, balanceTimeout),
+                sudo(sudoInterBtcAPI, async () => sudoInterBtcAPI.tokens.setBalance(vaultAccountId, freeBalanceToSet)),
+            ]);
+            await sudo(sudoInterBtcAPI, async () =>
```

Comment:
Is there any reason you're sudo-setting this balance twice?

Reply:
> Is there any reason you're sudo-setting this balance twice?

Missed that one after some refactoring. Thanks for spotting.